internal/kafka/producer_pool.go

```diff
@@ -0,0 +1,71 @@
+package kafka
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+
+	"github.com/gol4ng/logger"
+	kafkaconfluent "gopkg.in/confluentinc/confluent-kafka-go.v1/kafka"
+)
+
+type ProducerPool interface {
+	Close()
+	Produce(ctx context.Context, messages chan *Message)
+}
+
+type producerPool struct {
+	logger        logger.LoggerInterface
+	client        Client
```

Comment:
i think we can have the same interface signature between Client and ProducerPool
In fact the ProducerPool is just a decorator of a Producer but it will add the sub go routine Producer
so i think we can:
- rename `ProducerPool interface` in `Producer interface`
- transform the Client in Producer
what did you think about these changes?

Reply:
I agree this is the same interface, what about renaming client to producer and every client_* to producer_* and let producer_pool be one of them?